lang/java/avro/src/main/java/org/apache/avro/reflect/ReflectData.java

```diff
@@ -337,6 +350,33 @@ static Class getClassProp(Schema schema, String prop) {
     ARRAY_CLASSES.put(boolean.class, boolean[].class);
   }
 
+  /**
+   * It returns false for non-string-maps because Avro writes out such maps
+   * as an array of records. Even their JSON representation is an array.
+   */
+  protected boolean isMap(Object datum) {
```

Comment:
I'm afraid I misled you here. Because this is protected, it is part of the public API. I think `isStringMap` is a better name, but we need to make sure `isMap` remains. What about changing this back to `isMap` in generic, but adding `isStringMap` to reflect and having `isMap` call it? That way the change is limited to reflect and doesn't change the meaning of the API.

Sorry, my mistake!